### ISSUE TITLE
Separate `cheat_mode` from `debug_mode`

### ISSUE THREAD

**Issue by LordGolias** [LordGolias/Brytenwalda#14]
Currently the code uses `$cheat_mode` to display messages related to two different aspects of the game: 

* cheats: when the player gets advantages (e.g. where the companions are)
* debug: when the player is debugging a problem in the game

I believe this should be separated by creating a new global variable for debug mode, that can be modified in a menu (e.g. camp).

This way players that want to play in cheat mode are not spammed with debug messages.

**Comment by gdwitt** [comment]
Yes, you can see that the 2012 developers had about 3 levels of cheat modes in the source files. I was not sure how they worked so I conveted them all to 1. The reasoning is that players don't need cheat mode on. However, if we have time, it would be wise to add a level 2 cheat mode for debugging.
I think players are getting distracted by all the messaging. If you wish, I will research the BW source files to figure how they implemented 3 cheat mode levels.

**Comment by gdwitt** [comment]
The debug code is still alive an well in the code.
1) Level 1 is all cheat/debug messages. Use: ge, "$cheat_mode", 1 or gt, "$cheat_mode", 0.
I think the ge is safer because you see the 1 there.
2) Level 2 is for "Political messages only" Use ge, "$cheat_mode", 2.
3) Level 3 is for "Economic message only" Use ge, "$cheat_mode", 3
It would be good practice to start using 2 and 3 as 1 can see both of their messages.

You control this in-game by going to camp menu/take an action.
There you see these three options:
All messages
Political message only 
Economic messages

Here's a simple trigger that sets it to 2 then back to 1:
```
(try_begin),
		(ge, "$cheat_mode", 1),
	
		(try_for_range, ":king", "trp_kingdom_1_lord", "trp_knight_1_1"),		
			(store_add, ":proper_faction", ":king", "fac_kingdom_1"),
			(val_sub, ":proper_faction", "trp_kingdom_1_lord"),
			(store_faction_of_troop, ":actual_faction", ":king"),
			
			(neq, ":proper_faction", ":actual_faction"),
			(neq, ":actual_faction", "fac_commoners"),
			(ge, "$cheat_mode", 2),
			(neq, ":king", "trp_kingdom_2_lord"),
			
			(str_store_troop_name, s4, ":king"),
			(str_store_faction_name, s5, ":actual_faction"),
			(str_store_faction_name, s6, ":proper_faction"),
			(str_store_string, s65, "@{!}DEBUG - {s4} is in {s5}, should be in {s6}, disabling political cheat mode"),
			(display_message, "@{s65}"),
			(rest_for_hours, 0, 0, 0),
			
			(assign, "$cheat_mode", 1),
			(jump_to_menu, "mnu_debug_alert_from_s65"),
		(try_end),
```
